File: sdk/php/intellij/tty/UnImplTtyConnector.php
<?php

namespace php\intellij\tty;

class UnImplTtyConnector extends TtyConnector {

    /**
     * UnImplTtyConnector constructor.
     */
    public function __construct() {
        // Some java code ...
    }

    /**
     * @return void
     */
    public function close() {
        // Some java code ...
    }

    /**
     * @param array $termSize
     * @param array $pixelSize
     */
    public function resize(array $termSize, array $pixelSize) {
        // Some java code ...
    }

    /**
     * @return string
     */
    public function getName(): string {
        // Some java code ...
    }

    /**
     * @param string $buf
     * @param int $offset
     * @param int $length
     * @return int
     */
    public function read(string $buf, int $offset, int $length): int {
        // Some java code ...
    }

    /**
     * @param string $buf
     */
    public function write(string $buf) {
        // Some java code ...
    }

    /**
     * @return bool
     */
    public function isConnected(): bool {
        // Some java code ...
    }

    /**
     * @return int
     */
    public function waitFor(): int {
        // Some java code ...
    }
}

File: src-jvm/main/java/org/develnext/jphp/fx/jterminal/support/UnImplTtyConnector.java
package org.develnext.jphp.fx.jterminal.support;

import com.jediterm.terminal.Questioner;
import com.jediterm.terminal.TtyConnector;

import java.awt.*;
import java.io.IOException;

public class UnImplTtyConnector implements TtyConnector {
    @Override
    public boolean init(Questioner questioner) {
        return true;
    }

    @Override
    public void close() {

    }

    @Override
    public void resize(Dimension dimension, Dimension dimension1) {

    }

    @Override
    public String getName() {
        return "UnImpl";
    }

    @Override
    public int read(char[] chars, int i, int i1) throws IOException {
        return 0;
    }

    @Override
    public void write(byte[] bytes) throws IOException {

    }

    @Override
    public boolean isConnected() {
        return false;
    }

    @Override
    public void write(String s) throws IOException {

    }

    @Override
    public int waitFor() throws InterruptedException {
        return 0;
    }
}
